syntax = "proto3";
package audits;
service Audits {
  rpc GetAudits (GetAuditsRequest) returns (GetAuditsResponse) {}
  rpc GetAudit (GetAuditRequest) returns (GetAuditResponse) {}
  rpc CreateAudit (CreateAuditRequest) returns (CreateAuditResponse) {}
  rpc UpdateAudit (UpdateAuditRequest) returns (UpdateAuditResponse) {}
  rpc DeleteAudit (DeleteAuditRequest) returns (DeleteAuditResponse) {}
  
}

message Audit{
  string id = 1;
  string name = 2;
  string description = 3;
  string data = 4;
  string status = 5;
  string created_at = 6;
  string created_by = 7;
  string updated_at = 8;
  string updated_by = 9;
  string deleted_at = 10;
  string deleted_by = 11;
  string type = 12;

}
message GetAuditsRequest {
  string id = 1;
  string name = 2;
  string description = 3;
  string data = 4;
  string status = 5;
  string created_at = 6;
  string created_by = 7;
  string updated_at = 8;
  string updated_by = 9;
  string deleted_at = 10;
  string deleted_by = 11;
  string type = 12;
}
message GetAuditsResponse {
  repeated Audit audits = 1;
  int32 status = 2;
  string message = 3;
}
message GetAuditRequest {
  string id = 1;
}
message GetAuditResponse {
  Audit audit = 1;
  int32 status = 2;
  string message = 3;
}
message CreateAuditRequest {
  string name = 1;
  string description = 2;
  string data = 3;
  string status = 4;
  string created_at = 5;
  string created_by = 6;
  string updated_at = 7;
  string updated_by = 8;
  string deleted_at = 9;
  string deleted_by = 10;
  string type = 11;
}
message CreateAuditResponse {
  Audit audit = 1;
  int32 status = 2;
  string message = 3;
}
message UpdateAuditRequest {
  string id = 1;
  string name = 2;
  string description = 3;
  string data = 4;
  string status = 5;
  string created_at = 6;
  string created_by = 7;
  string updated_at = 8;
  string updated_by = 9;
  string deleted_at = 10;
  string deleted_by = 11;
  string type = 12;

}
message UpdateAuditResponse {
  Audit audit = 1;
  int32 status = 2;
  string message = 3;

}
message DeleteAuditRequest {
  string id = 1;
}
message DeleteAuditResponse {
  int32 status = 2;
  string message = 3;

}